splight_lib/models/alert.py

```diff
@@ -1,48 +1,207 @@
+        pattern = r"\$\w+"
+        args = re.findall(pattern, self.expression)
+        str_args = ", ".join(args)
+        body = f"function ({str_args}) {{ return {self.expression} }}"
+        expression_plain = {
+            "$function": {"body": body, "args": args, "lang": "js"}
+        }
+        return json.dumps(expression_plain)
+
+    def _get_query_plain(self):
+        query_plain = [
+            {
+                "$match": {
+                    "asset": self.query_filter_asset["id"],
+                    "attribute": self.query_filter_attribute["id"],
+                }
+            },
+        ]
+        if self.query_group_unit and self.query_group_function:
+            query_plain.extend(
+                [
+                    {
+                        "$addFields": {
+                            "timestamp": {
+                                "$dateTrunc": {
+                                    "date": "$timestamp",
+                                    "unit": self.query_group_unit,
+                                    "binSize": 1,
+                                }
+                            }
+                        }
+                    },
+                    {
+                        "$group": {
+                            "_id": "$timestamp",
+                            "value": {
+                                f"${self.query_group_function}": "$value"
+                            },
+                            "timestamp": {"$last": "$timestamp"},
+                        }
+                    },
+                ]
+            )
+        return json.dumps(query_plain)
+
+
+class AlertStatus(LowercaseStrEnum):
+    ALERT = auto()
+    NO_ALERT = auto()
+    NO_DATA = auto()
+    TIMEOUT = auto()
+    DISABLED = auto()
+    ERROR = auto()
+    WARNING = auto()
+
+
+class AlertType(LowercaseStrEnum):
+    CRON = auto()
+    RATE = auto()
+
+
+class StatementOperator(str, Enum):
     GREATER_THAN = "gt"
     GREATER_THAN_OR_EQUAL = "ge"
     LOWER_THAN = "lt"
     LOWER_THAN_OR_EQUAL = "le"
     EQUAL = "eq"
 
 
-class AlertVariableType(str, Enum):
-    STR = "str"
-    INT = "int"
-    FLOAT = "float"
+class ThresholdStatus(LowercaseStrEnum):
+    ALERT = auto()
+    WARNING = auto()
+    NO_ALERT = auto()
 
 
-class VariableDict(TypedDict):
-    name: str
-    type: str
-    value: Union[str, Dict]
+class AlertSeverity(LowercaseStrEnum):
+    SEV1 = auto()
+    SEV2 = auto()
+    SEV3 = auto()
+    SEV4 = auto()
+    SEV5 = auto()
+    SEV6 = auto()
+    SEV7 = auto()
+    SEV8 = auto()
 
 
-class AlertCondition(BaseModel):
-    id: Optional[str] = Field(None, max_length=100)
-    name: str
-    type: AlertVariableType = AlertVariableType.FLOAT
-    left_operand: str
-    right_operand: str
-    variables: List[VariableDict] = []
-    operator: AlertOperator = AlertOperator.EQUAL
+class AlertThreshold(BaseModel):
+    value: int
+    status: ThresholdStatus
+    status_text: Optional[str]
```

Comment:
```suggestion
    status_text: Optional[str] = None
```
Set the dafault value to avoid pydantic to fail. Use the same value as used in API

Reply:
👍🏻